common/common.go

```diff
@@ -436,27 +508,34 @@ func RunBuild(verbose bool, dir, imageName, dockerfile string, buildArgs []strin
 	}
 
 	go func(done chan<- error) {
-		args := []string{
-			"build",
-			"-t", imageName,
-			"-f", dockerfile,
-		}
-		if noCache {
-			args = append(args, "--no-cache")
-		}
+		var dockerBuildCmdArgs []string
+		// Depending whether architecture list is passed or not trigger docker buildx or docker build accordingly
+		if architectures != nil && len(architectures) != 0 {
+			err := initializeContainerBuilder(containerEngineType)
+			if err != nil {
+				done <- err
+				return
+			}
 
-		if len(buildArgs) > 0 {
-			for _, buildArg := range buildArgs {
-				args = append(args, "--build-arg", buildArg)
+			var mappedArchitectures []string
+			for _, arch := range architectures {
+				mappedArch, ok := architectureMap[arch]
+				if !ok {
+					err = errors.New(fmt.Sprintf("invalid architecture type : %v", arch))
+					done <- err
+					return
+				}
+				mappedArchitectures = append(mappedArchitectures, mappedArch)
 			}
+
+			dockerBuildCmdArgs = buildXDockerCommand(imageName, dockerfile,  buildArgs, noCache, mappedArchitectures)
+			// perform cleanup
+			defer cleanupContainerBuilder(containerEngineType)
```

Comment:
for 1 execution, we are cleaning up the docker driver twice (Once in Init and here). Why is it required here?

Reply:
1st one to remove any existing instance if customer had same name instance by any chance.
2nd one to ensure proper cleanup once command is done.